cmd/mqtt/main.go

```diff
@@ -122,6 +126,30 @@ func main() {
 		log.Fatalf(err.Error())
 	}
 
+	if cfg.mqttTargetHealthCheck != "" {
+		for {
+			res, err := http.Get(cfg.mqttTargetHealthCheck)
+			if err != nil {
+				logger.Info(fmt.Sprintf("Broker not ready: %s ", err.Error()))
+			}
+			if res != nil {
+				body, err := ioutil.ReadAll(res.Body)
+				if err != nil {
+					logger.Warn(fmt.Sprintf("Error reading response body: %s", err.Error()))
+				}
+				if err := res.Body.Close(); err != nil {
+					logger.Warn(fmt.Sprintf("Error closing response body: %s", err.Error()))
+				}
+				if res.StatusCode == http.StatusOK {
+					logger.Info("MQTT Broker health check successful")
+					break
+				}
+				logger.Info(fmt.Sprintf("Broker not ready, status code: %d, body: %s", res.StatusCode, body))
+			}
+			time.Sleep(healthCheckSleep)
```

Comment:
I would add a backoff to prevent this loop from running indefinitely:
```go
if i == maxHealthCheckRetries {
    logger.Error("MQTT healthcheck limit exceeded, exiting.")
    os.Exit(1)
}
time.Sleep(i * mqttHealthcheckSleep)
```

This can come in handy in the case of running service natively.

Reply:
I do not like this whole construct, it looks clunky.

Can you please use https://github.com/cenkalti/backoff and solve this with [`Retry()`](https://github.com/cenkalti/backoff/blob/c2975ffa541a1caeca5f76c396cb8c3e7b3bb5f8/example_test.go#L14) function.

Then if `Retry()` succeeds you can try reading the body, and if this fails just restart (that means broker is ready, but there were some more serious error. And this usually should not happen).

Reply:
Simply extracting to a separate function and calling it here will help.

Reply:
I would like that we try with https://github.com/cenkalti/backoff in order to prepare terrain for the eventual changing of other services.

Reply:
switched to cenkalti/backoff